tests/test_adjointA.py

```diff
@@ -30,3 +33,30 @@ def test_acoustic(dimensions, time_order, space_order, fix_dim):
     term2 = linalg.norm(rec.data) ** 2
     print(term1, term2, ("%12.12f") % (term1 - term2), term1 / term2)
     assert np.isclose(term1 / term2, 1.0, atol=0.001)
+
+
+@pytest.mark.parametrize('dimensions', [(60, 70), (60, 70, 80)])
+@pytest.mark.parametrize('fix_dim', [True, False])
+def test_acoustic_constant(dimensions, fix_dim):
+    solver = setup_c(dimensions=dimensions, time_order=2,
+                     space_order=8, nbpml=14)
+    srca = PointSource(name='srca', ntime=solver.source.nt,
+                       coordinates=solver.source.coordinates.data)
+
+    if fix_dim:
+        time.size = solver.source.nt
+    else:
+        time.size = None
+    # Run forward and adjoint operators
+    rec, _, _ = solver.forward(save=False)
+    solver.adjoint(rec=rec, srca=srca)
+
+    # Actual adjoint test
+    term1 = np.dot(srca.data.reshape(-1), solver.source.data)
+    term2 = linalg.norm(rec.data) ** 2
+    print(term1, term2, ("%12.12f") % (term1 - term2), term1 / term2)
+    assert np.isclose(term1 / term2, 1.0, atol=0.001)
+
+
+if __name__ == "__main__":
```

Comment:
You can easily run these with `py.test .... -k <some patter to identify single test>`. Using this instead of `__main__` functions means that you also etst your on test setup when you do dev runs, and pytest bugs like the silent `setup()` would be caught straight away...